Sources/DockProgress/DockProgress.swift

```diff
@@ -1,13 +1,34 @@
 import Cocoa
+import CoreVideo
 
 @MainActor
 public enum DockProgress {
-	private static var previousProgress: Double = 0
+    private static var t: Double = 0;
+    private static var displayLink: CVDisplayLink?
 	private static var progressObserver: NSKeyValueObservation?
 	private static var finishedObserver: NSKeyValueObservation?
 
 	private static let dockContentView = with(ContentView()) {
 		NSApp.dockTile.contentView = $0
+        
+        let displayLinkCallback: CVDisplayLinkOutputCallback = {(displayLink: CVDisplayLink, inNow: UnsafePointer<CVTimeStamp>, inOutputTime: UnsafePointer<CVTimeStamp>, flagsIn: CVOptionFlags, flagsOut: UnsafeMutablePointer<CVOptionFlags>, displayLinkContext: UnsafeMutableRawPointer?) -> CVReturn in
```

Comment:
It would be better to wrap up the display link logic into a simple decoupled class, for maintainability.